### PR TITLE
Pass JdbcIdentity to DriverConnectionFactory#openConnection

### PR DESCRIPTION
**Pull request by kokosing**
Pass JdbcIdentity to DriverConnectionFactory#openConnection

### CODE REVIEW

**Comment by dain** [comment]
Just curious, what is the motivation for this change.

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcIdentity.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public int hashCode()
+    {
+        return Objects.hash(extraCredentials);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("extraCredentials", extraCredentials)
```
Comment (electrum) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcIdentity.java]: I think we only want to include the keys here, like in `Identity`, as the values could be passwords or other sensitive things.

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcIdentity.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import static java.util.Objects.requireNonNull;
+
+public class JdbcIdentity
+{
+    public static JdbcIdentity from(ConnectorSession session)
+    {
+        return from(session.getIdentity());
+    }
+
+    public static JdbcIdentity from(ConnectorIdentity connectorIdentity)
```
Comment (electrum) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcIdentity.java]: This seems not to be used.

**Comment by dain** [comment]
@cla-bot check

**Comment by kokosing** [comment]
>  Why not pass `ConnectorIdentity`?

There are two reasons:
 - `ConnectorIdentity` has more information than I need, it contains (roles) and hence it is difficult to write proper equals/hashCode methods, which could needed to implement connection pooling (caching).
 - `JdbcConnectorIdentity` allows to pass different set information than `ConnectorIdentity` has, i.e. allows to pass additional, custom connector based information

**Comment by kokosing** [comment]
> Just curious, what is the motivation for this change.

Thanks to this change connector is capable to open different JDBC connection depending on `ConnectorIdentity`. User impersonation.

File: presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestJdbcClient.java [hunk 3]
```diff
@@ -65,17 +65,17 @@ public void tearDown()
     @Test
     public void testMetadata()
     {
-        assertTrue(jdbcClient.getSchemaNames().containsAll(ImmutableSet.of("example", "tpch")));
-        assertEquals(jdbcClient.getTableNames("example"), ImmutableList.of(
+        assertTrue(jdbcClient.getSchemaNames(JdbcIdentity.from(session)).containsAll(ImmutableSet.of("example", "tpch")));
```
Comment (findepi) [inline presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestJdbcClient.java]: nit: extract `JdbcIdentity.from(session)` as var

File: presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestJdbcRecordSetProvider.java [hunk 4]
```diff
@@ -47,6 +47,7 @@
 @Test
 public class TestJdbcRecordSetProvider
 {
+    private static final JdbcIdentity IDENTITY = new JdbcIdentity("user", ImmutableMap.of());
```
Comment (findepi) [inline presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestJdbcRecordSetProvider.java]: nit: empty line after class-level consts

```suggestion
    private static final JdbcIdentity IDENTITY = new JdbcIdentity("user", ImmutableMap.of());

```

File: presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestingDatabase.java [hunk 5]
```diff
@@ -96,15 +96,15 @@ public JdbcClient getJdbcClient()
 
     public JdbcSplit getSplit(String schemaName, String tableName)
     {
-        JdbcTableHandle jdbcTableHandle = jdbcClient.getTableHandle(new SchemaTableName(schemaName, tableName));
+        JdbcTableHandle jdbcTableHandle = jdbcClient.getTableHandle(JdbcIdentity.from(session), new SchemaTableName(schemaName, tableName));
         JdbcTableLayoutHandle jdbcLayoutHandle = new JdbcTableLayoutHandle(jdbcTableHandle, TupleDomain.all());
-        ConnectorSplitSource splits = jdbcClient.getSplits(jdbcLayoutHandle);
+        ConnectorSplitSource splits = jdbcClient.getSplits(JdbcIdentity.from(session), jdbcLayoutHandle);
```
Comment (findepi) [inline presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestingDatabase.java]: extract `JdbcIdentity.from(session)`

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcIdentity.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    public int hashCode()
+    {
+        return Objects.hash(user, extraCredentials);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("extraCredentials", extraCredentials.keySet())
```
Comment (electrum) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcIdentity.java]: Add `user` here

**Pull request merged**